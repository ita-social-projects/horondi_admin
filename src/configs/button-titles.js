const buttonTitles = {
  DELETE_TITLE: 'Видалити',
  EDIT_TITLE: 'Редагувати',
  CREATE_NEWS_TITLE: 'Додати новину',
  CREATE_PATTERN_TITLE: 'Додати гобелен',
  CREATE_MODEL_TITLE: 'Додати модель',
  MODEL_SAVE_TITLE: 'Зберегти',
  MODEL_CONSTRUCTOR: 'Конструктор',
  CANCEL_TITLE: 'Ні',
  LOGOUT_TITLE: 'Вихід',
  CREATE_BUSINESS_PAGE: 'Додати бізнес сторінку',
  CREATE_CONTACT_TITLE: 'Додати контакти',
  SWITCH_USER_STATUS_TITLE: 'Змінити статус користувача',
  USER_INACTIVE_TITLE: 'Деактивувати',
  USER_ACTIVE_TITLE: 'Активувати',
  ADD_CATEGORY: 'Додати категорію',
  GO_BACK_TITLE: 'Назад',
  ADD_CATEGORY_IMAGE: 'Додати зображення',
  ADD_CATEGORY_NAME: 'Додати назву',
  CANCEL: 'Відмінити',
  SAVE_CATEGORY: 'Зберегти категорію',
  CREATE_SPECIAL_USER: 'Створити спецкористувача',
  ADD_PHOTO_LABEL: '+',
  CREATE_CATEGORY: 'Створити категорію',
  PATTERN_REMOVE_MESSAGE: 'Ви впевнені, що хочете видалити цей гобелен?',
  REMOVE_CONTACT_MESSAGE: 'Ви впевнені, що хочете видалити цей контакт?',
  USER_UNACTIVE_TITLE: 'Деактивувати',
  SHOW_COMMENTS_TITLE: 'Переглянути коментарі',
  HIDE_COMMENTS_TITLE: 'Приховати коментарі',
  SAVE_ORDER: 'Зберегти замовлення',
  SAVE_TITLE: 'Зберегти',
  UNDERSTAND: 'Зрозуміло',
  ANSWER: 'Відповісти',
  TO_SPAM: 'У СПАМ',
  MOVE_ALL_TO_SPAM: 'Перемістити у СПАМ',
  CREATE_SIZE_TITLE: 'Створити розмір',
  SAVE_SIZE_TITLE: 'Зберегти розмір',
  CREATE_MATERIAL_TITLE: 'Створити матеріал',
  CREATE_COLOR_TITLE: 'Створити колір',
  UPDATE_COLOR_TITLE: 'Зберегти зміни',
  SAVE_MATERIAL: 'Зберегти матеріал',
  CREATE_HEADER_TITLE: 'Створити посилання',
  HEADER_SAVE_TITLE: 'Зберегти посилання',
  GO_TO_MATERIAL_COLOR_PALLET: 'Редагувати палітру',
  CLOSE_DIALOG_TITLE: 'Закрити вікно',
  ORDER_DETAILS: 'Деталі',
  USER_STATUS_TITLE: 'Статус',
  CLEAR_FILTERS: 'Очистити фільтри',
  CREATE_SLIDE_TITLE: 'Створити слайд',
  OPEN_SLIDE_EDIT: 'Увімкнути редагування',
  SAVE_SLIDE_ORDER: 'Зберегти зміни',
  CANCEL_SLIDE_ORDER: 'Скасувати зміни',
  AUTHOR_PHOTO: 'Фото Автора',
  MAIN_PHOTO: 'Головне Фото',
  CREATE_CONSTRUCTOR_BASIC_TITLE: 'Додати основу',
  CREATE_CONSTRUCTOR_FRONT_POCKET_TITLE: 'Додати передню кишеню',
  CREATE_CONSTRUCTOR_BOTTOM_TITLE: 'Додати низ',
  GO: 'Перейти',
  CATEGORY_TITLE: 'Категорія',
  LOOK: 'Переглянути',
  ADD_ORDER: 'Додати замовлення',
  USER_ROLE_TITLE: 'Роль',
  EVENT_TITLE: 'Подія',
  SEND_CONFIRM: 'Надіслати підтвердження',
  SEND_CODE: 'Надіслати код для підтвердження',
  EXIT_WITHOUT_SAVING: 'Вийти без збереження',
  ORDER_STATUS: 'Статус замовлення',
  PAYMENT_STATUS: 'Статус оплати',
  SIZE_NAME: 'Розмір',
  AVAILABLE: 'Доступність',
  PURPOSE: 'Застосування',
  CATEGORY: 'Категорії',
  PATTERN: 'Гобелени',
  MODEL: 'Моделі',
  YES_BUTTON: 'Так',
  NO_BUTTON: 'Ні',
  CREATE_BACK_TITLE: 'Додати спинку',
  CREATE_POCKETS_TITLE: 'Додати кишеню',
  PATTERN_AVAILABLE: 'Доступний',
  PATTERN_MATERIAL: 'Матеріал',
  CREATE_CONSTRUCTOR: 'Створити новий '
};

export default buttonTitles;
