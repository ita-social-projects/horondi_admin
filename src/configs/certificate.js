const certificate = {
  TITLE_CERTIFICATE: 'HORONDI',
  CERTIFICATE_TEXT_WITH_VALUE: (value) =>
    `Подарунковий сертифікат на ${value} грн`
};

export default certificate;
