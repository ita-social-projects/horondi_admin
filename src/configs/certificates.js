const certificates = {
  certificateStatus: {
    ACTIVE: 'Активний',
    USED: 'Використаний',
    EXPIRED: 'Протермінований'
  }
};

export default certificates;
