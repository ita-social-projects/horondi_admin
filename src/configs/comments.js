const comments = {
  commentType: {
    COMMENT: 'Коментар',
    REPLY_COMMENT: 'Відповідь'
  }
};

export default comments;
