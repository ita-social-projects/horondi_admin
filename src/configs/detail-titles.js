const detailTitles = {
  users: {
    avatar: { id: 'avatar' },
    name: { id: 'name' },
    status: { id: 'status' },
    sections: [
      { id: 'phone', label: 'Телефон' },
      { id: 'country', label: 'Країна' },
      { id: 'city', label: 'Місто' },
      { id: 'adress', label: 'Адреса' },
      { id: 'postCode', label: 'Поштовий індекс' }
    ]
  },
  emailQuestions: {
    customer: {
      date: 'Дата запитання:',
      status: 'Статус:',
      sender: 'Відправник:',
      email: 'Email:',
      question: 'Запитання:'
    },
    admin: {
      date: 'Дата відповіді:',
      admin: 'Адмін:',
      answer: 'Відповідь:'
    }
  }
};

export default detailTitles;
