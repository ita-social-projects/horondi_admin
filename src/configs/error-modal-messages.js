export const constructorErrors = {
  BASIC_NOT_FOUND: 'Основу не знайдено!',
  BASIC_ALREADY_EXIST: 'Така основа вже існує!',
  CONSTRUCTOR_BOTTOM_NOT_FOUND: 'Низ не знайдено!',
  CONSTRUCTOR_BOTTOM_ALREADY_EXIST: 'Такий низ вже існує!',
  FRONT_POCKET_NOT_FOUND: 'Кишеню не знайдено!',
  FRONT_POCKET_ALREADY_EXIST: 'Така кишеня вже існує!',
  BACK_NOT_FOUND: 'Спинку не знайдено!',
  BACK_ALREADY_EXIST: 'Така спинка вже існує!',
  CONSTRUCTOR_NOT_FOUND: 'Конструктор не знайдено',
  CONSTRUCTOR_ALREADY_EXIST: 'Конструктор вже існує'
};

export const businessPageErrors = {
  BUSINESS_TEXT_NOT_FOUND: 'Сторінку не знайдено!',
  BUSINESS_TEXT_WITH_THIS_CODE_ALREADY_EXIST: 'Така сторінка вже існує!'
};

export const bottomTranslations = {
  BOTTOM_NOT_FOUND: 'Низ не знайдено!',
  BOTTOM_ALREADY_EXIST: 'Такий низ вже існує!'
};

export const backTranslations = {
  BACK_NOT_FOUND: 'Спинку не знайдено!',
  BACK_ALREADY_EXIST: 'Така спинка вже існує!'
};

export const categoryErrors = {
  CATEGORY_NOT_FOUND: 'Категорію не знайдено!',
  CATEGORY_ALREADY_EXIST: 'Така категорія вже існує!'
};

export const colorsTranslations = {
  COLOR_NOT_FOUND: 'Колір не знайдено!',
  COLOR_ALREADY_EXIST: 'Такий колір вже існує!'
};

export const contactTranslations = {
  CONTACT_NOT_FOUND: 'Контакт не знайдено!',
  CONTACT_ALREADY_EXIST: 'Контакт з такими даними вже існує!'
};

export const headerTranslations = {
  HEADER_NOT_FOUND: 'Посилання не знайдено!',
  HEADER_ALREADY_EXIST: 'Таке посилання вже існує!'
};

export const sizesTranslations = {
  SIZE_NOT_FOUND: 'Розмір не знайдено!',
  SIZE_ALREADY_EXIST: 'Такий розмір вже існує!'
};

export const modelTranslations = {
  MODEL_NOT_FOUND: 'Модель не знайдено!',
  MODEL_ALREADY_EXIST: 'Така модель вже існує!'
};

export const newsErrors = {
  NEWS_NOT_FOUND: 'Новину не знайдено!',
  NEWS_ALREADY_EXIST: 'Така новина вже існує!'
};

export const patternTranlations = {
  PATTERN_NOT_FOUND: 'Гобелен не знайдено!',
  PATTERN_ALREADY_EXIST: 'Такий гобелен вже існує!'
};

export const homePageEditErrors = {
  HOME_PAGE_DATA_NOT_FOUND: 'Дані головної сторінки не знайдено!',
  HOME_PAGE_DATA_ALREADY_EXIST: 'На сторінці вже наявні такі дані!'
};

export const commentsErrors = {
  COMMENT_NOT_FOUND: 'Коментар не знайдено!',
  COMMENG_ALREADY_EXIST: 'Такий коментар вже існує!',
  REPLY_COMMENT_IS_NOT_PRESENT: 'Коментар не існує!'
};

export const slidesTranslations = {
  SLIDE_NOT_FOUND: 'Слайд не знайдено!',
  SLIDE_ALREADY_EXIST: 'Слайд з таким заголовком вже існує!',
  IMAGE_NOT_PROVIDED: 'Відсутнє зображення'
};

export const materialTranslations = {
  MATERIAL_NOT_FOUND: 'Матеріал не знайдено!',
  MATERIAL_ALREADY_EXIST: 'Такий матеріал вже існує!'
};

export const productsErrors = {
  PRODUCT_NOT_FOUND: 'Продуктів не знайдено',
  PRODUCT_ALREADY_EXIST: 'Такий продукт вже існує!'
};

export const userTranslations = {
  USER_IS_BLOCKED: 'Користувача заблоковано',
  WRONG_CREDENTIALS: 'Неправильний логін або пароль',
  INVALID_PERMISSIONS: 'Недостатньо прав для доступу',
  USER_NOT_FOUND: 'Користувача не знайдено'
};

export const errorStatuses = {
  ERROR_PAGE_STATUS: 'Сторінку не знайдено!',
  ERROR_BOUNDARY_STATUS: 'Сталася помилка!'
};

export const usersErrors = {
  USER_NOT_FOUND: 'Користувача не знайдено'
};

export const adminsErrors = {
  ADMIN_NOT_FOUND: 'Адміністратора не знайдено'
};
