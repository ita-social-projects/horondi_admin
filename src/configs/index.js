import { colors } from '@material-ui/core';
import routes from './routes';
import {
  menuCategories,
  clientMenuCategories,
  catalogMenuCategories,
  staticPagesCategories,
  constructorMenuCategories
} from './menu-categories';
import tableHeadRowTitles from './table-head-row-titles';
import detailTitles from './detail-titles';
import buttonTitles from './button-titles';
import statuses from './statuses';
import filterLabels from './filter-labels';
import materialUiConstants from './material-ui-constants';
import {
  loginErrorMessages,
  colorErrorMessages,
  contactErrorMessages,
  patternErrorMessages,
  backErrorMessages,
  materialErrorMessages,
  newsErrorMessages,
  errorMessages,
  categoryErrorMessages,
  modelErrorMessages,
  headerErrorMessages,
  homePageSlideErrorMessages,
  constructorErrorMessages,
  statsErrorMessages,
  paginationInputErrorMessages,
  commentErrorMessages,
  replyCommentErrorMessages,
  sizeErrorMessages,
  businessPageErrorMessages,
  pocketsErrorMessages
} from './error-messages';
import messages from './messages';
import formRegExp from './form-regexp';
import labels from './labels';
import { newsPaginationPayload, contactsPaginationPayload } from './pagination';
import titles from './titles';
import languages from './languages';
import {
  imageUploadNewsInputsId,
  valueKeys,
  inputNames
} from './news-form-helpers';

export const config = {
  fetchPolicy: 'no-cache',
  app: {
    title: 'Horondi Admin Portal',
    serverUrl: 'http://localhost:5000/',
    drawerWidth: 230,
    snackBarDuration: 4000,
    rowsPerPageOptions: [10, 25, 50, 100]
  },
  languages,
  tokenName: 'HORONDI_AUTH_TOKEN',
  templates: {
    categoryTemplate: {
      available: false,
      code: '',
      images: {
        large: '',
        medium: '',
        small: '',
        thumbnail: ''
      },
      name: []
    }
  },
  userRole: {
    user: 'user',
    admin: 'admin',
    superadmin: 'superadmin'
  },
  userRoles: [
    { role: 'user', label: 'Юзер' },
    { role: 'admin', label: 'Адмін' },
    { role: 'superadmin', label: 'Суперадмін' }
  ],
  allowedforRegistrationRoles: ['admin', 'superadmin'],
  tabNames: {
    users: ['Користувачі', 'Адміністратори']
  },
  tableSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'medium'
  },
  iconSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'default'
  },
  colorCircleSizes: {
    SMALL_CIRCLE: '25px',
    DEFAULT_CIRCLE: '50px'
  },
  theme: {
    DARK_THEME: 'dark',
    LIGHT_THEME: 'light'
  },
  forbiddenRolesFromDeleting: ['superadmin'],
  dialogFormTitles: {
    REGISTER_ADMIN: 'Створити спецкористувача'
  },
  paginationPayload: {
    skip: 0,
    limit: 5,
    countPerPage: 6
  },
  sizesAvailableVariants: {
    AVAILABLE_TEXT: 'Доступний',
    UNAVAILABLE_TEXT: 'Відсутній'
  },
  pocketsAvailableVariants: {
    AVAILABLE_TEXT: 'Доступний',
    UNAVAILABLE_TEXT: 'Недоступний'
  },

  IMG_URL: 'https://horondi.blob.core.windows.net/horondi/images/',
  newsPerPage: 6,
  product: {
    sortBySelectOptions: [
      {
        label: 'Популярністю',
        value: 'popularity'
      },
      {
        label: 'Від дорогих до дешевих',
        value: 'sortDesc'
      },
      {
        label: 'Від дешевих до дорогих',
        value: 'sortAsc'
      },
      {
        label: 'Рейтингом',
        value: 'rate'
      }
    ],
    responsive: {
      superLargeDesktop: {
        breakpoint: { max: 4000, min: 3000 },
        items: 1
      },
      desktop: {
        breakpoint: { max: 3000, min: 1024 },
        items: 1
      },
      tablet: {
        breakpoint: { max: 1146, min: 464 },
        items: 1
      },
      mobile: {
        breakpoint: { max: 810, min: 0 },
        items: 1
      }
    }
  },
  popularity: 'popularity',
  rate: 'rate',
  sortAsc: 'sortAsc',
  sortDesc: 'sortDesc',
  submitKey: 'Enter',
  imagePrefix: 'https://horondi.blob.core.windows.net/horondi/images/',
  initialLanguageValues: [
    {
      lang: 'ua',
      value: ''
    },
    {
      lang: 'en',
      value: ''
    }
  ],
  doughnut: {
    colors: [
      colors.green[500],
      colors.red[700],
      colors.amber[600],
      colors.indigo[500]
    ],
    initialValues: {
      names: [],
      counts: [],
      relations: [],
      total: null
    }
  },
  bar: {
    initialValues: {
      labels: [],
      counts: []
    }
  },
  UAH: 'грн',
  maxItemsPerPage: 10,
  buttonTitles,
  detailTitles,
  messages,
  menuCategories,
  clientMenuCategories,
  catalogMenuCategories,
  constructorMenuCategories,
  staticPagesCategories,
  routes,
  statuses,
  tableHeadRowTitles,
  formRegExp,
  labels,
  titles,
  loginErrorMessages,
  colorErrorMessages,
  contactErrorMessages,
  categoryErrorMessages,
  patternErrorMessages,
  backErrorMessages,
  materialErrorMessages,
  newsErrorMessages,
  errorMessages,
  newsPaginationPayload,
  contactsPaginationPayload,
  modelErrorMessages,
  headerErrorMessages,
  statsErrorMessages,
  homePageSlideErrorMessages,
  constructorErrorMessages,
  paginationInputErrorMessages,
  commentErrorMessages,
  replyCommentErrorMessages,
  filterLabels,
  materialUiConstants,
  sizeErrorMessages,
  businessPageErrorMessages,
  imageUploadNewsInputsId,
  valueKeys,
  inputNames,
  pocketsErrorMessages
};
export const inputTypes = {
  button: 'button',
  submit: 'submit'
};

export const badgePosition = {
  vertical: 'top',
  horizontal: 'left'
};

export const dateFormat = ' HH:mm DD/MM/YYYY ';
export const dateFormatOrder = ' DD.MM.YYYY ';

export const formConstants = {
  formTypeSelect: 'select',
  isRequired: '*',
  notRequired: '',
  textFieldFilled: 'filled'
};
