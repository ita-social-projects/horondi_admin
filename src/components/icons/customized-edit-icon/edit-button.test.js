import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import EditButton from './customized-edit-icon';
import { config } from '../../../configs';

Enzyme.configure({ adapter: new Adapter() });

const { EDIT_TITLE } = config.buttonTitles;

describe('edit button tests', () => {
  const size = 'small';
  const mockCallBack = jest.fn();

  let component;

  beforeEach(() => {
    component = mount(<EditButton size={size} onClickHandler={mockCallBack} />);
  });
  afterEach(() => {
    component.unmount();
  });
  it('should click', () => {
    expect(component).toMatchSnapshot();
    expect(mockCallBack.mock.calls.length).toBe(0);
    component.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toBe(1);
  });
  it('should exist and have value', () => {
    expect(component.exists());
    expect(component.find('button').prop('aria-label')).toEqual(EDIT_TITLE);
    expect(component.find('button').prop('title')).toEqual(EDIT_TITLE);
    expect(component.prop('onClickHandler')).toEqual(mockCallBack);
  });
  it('should have type', () => {
    expect(component.find('button').type()).toEqual('button');
  });
  it('should have props', () => {
    expect(component.props().size).toBeDefined();
    expect(component.props().size).toEqual(size);
    expect(component.props().onClickHandler).toBeDefined();
    expect(component.props().onClickHandler).toBeTruthy();
  });
  it('should have default props', () => {
    expect(EditButton.propTypes.size).toBeDefined();
    expect(EditButton.propTypes.onClickHandler).toBeDefined();
  });
});
