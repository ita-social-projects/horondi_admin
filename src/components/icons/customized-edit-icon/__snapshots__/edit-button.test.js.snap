// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit button tests should click 1`] = `
<CustomizedEditIcon
  onClickHandler={[MockFunction]}
  size="small"
>
  <WithStyles(ForwardRef(Tooltip))
    title="Редагувати"
  >
    <ForwardRef(Tooltip)
      classes={
        Object {
          "arrow": "MuiTooltip-arrow",
          "popper": "MuiTooltip-popper",
          "popperArrow": "MuiTooltip-popperArrow",
          "popperInteractive": "MuiTooltip-popperInteractive",
          "tooltip": "MuiTooltip-tooltip",
          "tooltipArrow": "MuiTooltip-tooltipArrow",
          "tooltipPlacementBottom": "MuiTooltip-tooltipPlacementBottom",
          "tooltipPlacementLeft": "MuiTooltip-tooltipPlacementLeft",
          "tooltipPlacementRight": "MuiTooltip-tooltipPlacementRight",
          "tooltipPlacementTop": "MuiTooltip-tooltipPlacementTop",
          "touch": "MuiTooltip-touch",
        }
      }
      title="Редагувати"
    >
      <WithStyles(ForwardRef(IconButton))
        aria-describedby={null}
        aria-label="Редагувати"
        className=""
        color="secondary"
        data-cy="edit-btn"
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Редагувати"
      >
        <ForwardRef(IconButton)
          aria-describedby={null}
          aria-label="Редагувати"
          className=""
          classes={
            Object {
              "colorInherit": "MuiIconButton-colorInherit",
              "colorPrimary": "MuiIconButton-colorPrimary",
              "colorSecondary": "MuiIconButton-colorSecondary",
              "disabled": "Mui-disabled",
              "edgeEnd": "MuiIconButton-edgeEnd",
              "edgeStart": "MuiIconButton-edgeStart",
              "label": "MuiIconButton-label",
              "root": "MuiIconButton-root",
              "sizeSmall": "MuiIconButton-sizeSmall",
            }
          }
          color="secondary"
          data-cy="edit-btn"
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Редагувати"
        >
          <WithStyles(ForwardRef(ButtonBase))
            aria-describedby={null}
            aria-label="Редагувати"
            centerRipple={true}
            className="MuiIconButton-root MuiIconButton-colorSecondary"
            data-cy="edit-btn"
            disabled={false}
            focusRipple={true}
            onBlur={[Function]}
            onClick={[MockFunction]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            title="Редагувати"
          >
            <ForwardRef(ButtonBase)
              aria-describedby={null}
              aria-label="Редагувати"
              centerRipple={true}
              className="MuiIconButton-root MuiIconButton-colorSecondary"
              classes={
                Object {
                  "disabled": "Mui-disabled",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiButtonBase-root",
                }
              }
              data-cy="edit-btn"
              disabled={false}
              focusRipple={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="Редагувати"
            >
              <button
                aria-describedby={null}
                aria-label="Редагувати"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
                data-cy="edit-btn"
                disabled={false}
                onBlur={[Function]}
                onClick={[MockFunction]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                title="Редагувати"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <ForwardRef(EditIcon)
                    fontSize="small"
                  >
                    <WithStyles(ForwardRef(SvgIcon))
                      fontSize="small"
                    >
                      <ForwardRef(SvgIcon)
                        classes={
                          Object {
                            "colorAction": "MuiSvgIcon-colorAction",
                            "colorDisabled": "MuiSvgIcon-colorDisabled",
                            "colorError": "MuiSvgIcon-colorError",
                            "colorPrimary": "MuiSvgIcon-colorPrimary",
                            "colorSecondary": "MuiSvgIcon-colorSecondary",
                            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                            "root": "MuiSvgIcon-root",
                          }
                        }
                        fontSize="small"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                          />
                        </svg>
                      </ForwardRef(SvgIcon)>
                    </WithStyles(ForwardRef(SvgIcon))>
                  </ForwardRef(EditIcon)>
                </span>
                <WithStyles(memo)
                  center={true}
                >
                  <ForwardRef(TouchRipple)
                    center={true}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child",
                        "childLeaving": "MuiTouchRipple-childLeaving",
                        "childPulsate": "MuiTouchRipple-childPulsate",
                        "ripple": "MuiTouchRipple-ripple",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                        "root": "MuiTouchRipple-root",
                      }
                    }
                  >
                    <span
                      className="MuiTouchRipple-root"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component={null}
                        exit={true}
                      />
                    </span>
                  </ForwardRef(TouchRipple)>
                </WithStyles(memo)>
              </button>
            </ForwardRef(ButtonBase)>
          </WithStyles(ForwardRef(ButtonBase))>
        </ForwardRef(IconButton)>
      </WithStyles(ForwardRef(IconButton))>
      <ForwardRef(Popper)
        anchorEl={
          <button
            aria-label="Редагувати"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
            data-cy="edit-btn"
            tabindex="0"
            title="Редагувати"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        }
        className="MuiTooltip-popper"
        id={null}
        open={false}
        placement="bottom"
        popperOptions={
          Object {
            "modifiers": Object {
              "arrow": Object {
                "element": null,
                "enabled": false,
              },
            },
          }
        }
        transition={true}
      />
    </ForwardRef(Tooltip)>
  </WithStyles(ForwardRef(Tooltip))>
</CustomizedEditIcon>
`;
